### PR TITLE
elektra-web structure decision

### PR DESCRIPTION
**Pull request by omnidan**
as requested in #999

### CODE REVIEW

**Comment by markus2330** [comment]
Thank you! Unfortunately, we did not have time to discuss it: You argue well why we need a single point of entry (only Assumptions and Constraints are a bit mixed up), but you are missing on explaining why not have recursive elektrad/clusterd/some thing else not existing yet (with a single clusterd for access)

**Comment by markus2330** [comment]
Btw. the recursive decision please separately, but link between them. Also please link from README.md.

File: doc/decisions/elektra_web.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
+# elektra-web structure
```
Comment (markus2330) [inline doc/decisions/elektra_web.md]: Please add your decision in the README.md

File: doc/decisions/elektra_web.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+## Issue
+
+For elektra-web, there needs to be a way to remotely manage instances and groups
+of instances (clusters). The remote configuration of a single instance is
```
Comment (markus2330) [inline doc/decisions/elektra_web.md]: This should only describe the issue, not a solution.

File: doc/decisions/elektra_web.md (hunk 1 above)
Comment (omnidan) [inline doc/decisions/elektra_web.md]: In [this README.md](https://github.com/ElektraInitiative/libelektra/blob/master/src/tools/web/README.md)? I can add a link once this PR gets merged.

**Comment by omnidan** [comment]
@markus2330 I have updated this PR and created another one for the recursive decision (#1354)

File: doc/decisions/elektra_web.md (hunk 1 above)
Comment (markus2330) [inline doc/decisions/elektra_web.md]: No in the doc/decisions/README.md

File: doc/decisions/elektra_web.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+because the user would have to manually set up a new daemon for each cluster.
+Another problem with this approach is that we still don't have a single point of
+entry, so we would have to store information to connect to the daemons on the
+client.
+
+Using one daemon to manage all instances and clusters is the easiest solution,
+that way the client simply connects to that daemon, which forwards requests to
+single instances or multiple instances at once. In this case, the cluster daemon
+can also serve the client, further simplifying the whole structure, because we
+don't need to host the client on a separate web server.
```
Comment (markus2330) [inline doc/decisions/elektra_web.md]: Another argument pro single access is: Thus both clusterd and elektrad are nodejs, it does not matter dependency-wise. And a wrapper script can start both daemons easily.

The script should also setup the clusterd to have the elektrad on the same node as only instance added. This solution would be quite optimal.

File: doc/decisions/elektra_web.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+## Related decisions
+
+- [elektra-web recursive structure decision](elektra_web_recursive.md)
+
+## Notes
+
+There are still some other issues, like possible conflicts when adding an
+instance multiple times, or assigning an instance to multiple clusters.
+
+The cluster daemon should prevent adding the same instance twice. It also won't
```
Comment (markus2330) [inline doc/decisions/elektra_web.md]: This is also an argument and not notes. Notes would be a link to this PR and other discussions at github.

**Comment by markus2330** [comment]
Please also rebase this PR, it is quite old.

**Comment by omnidan** [comment]
@markus2330 I adjusted the decision according to your comments, rebased and added links to both decisions in the README.md.

**Comment by markus2330** [comment]
Something went wrong with rebasing, please cherry-pick the relevant commits.

**Comment by omnidan** [comment]
@markus2330 not sure what went wrong here. I made a new PR: #1383

**Pull request closed without merging**